dbms/src/Storages/Transaction/ApplySnapshot.cpp

```diff
@@ -387,11 +390,18 @@ std::vector<DM::ExternalDTFileInfo> KVStore::preHandleSSTsToDTFiles(
                 job_type,
                 /* split_after_rows */ global_settings.dt_segment_limit_rows,
                 /* split_after_size */ global_settings.dt_segment_limit_size,
+                region_id,
+                prehandle_task,
                 context);
 
             stream->writePrefix();
             stream->write();
             stream->writeSuffix();
+            if (stream->isAbort())
+            {
+                LOG_INFO(log, "Cancel stream because of upper layer abort [region_id={}]", region_id);
+                stream->cancel();
+            }
```

Comment:
Now we call `stream->cancel()` if the pre-handle is aborted, do we still need the `releasePreHandleSnapshot` method?

Reply:
Yes. There are cases when the pre handling has finished before proxy calls "abort". In this case, proxy have to call "release" here.